src/command.rs

```diff
@@ -145,111 +151,202 @@ pub fn run_wasm_pack(command: Command, log: &Logger) -> result::Result<(), Error
 // quicli::prelude::* imports a different result struct which gets
 // precedence over the std::result::Result, so have had to specify
 // the correct type here.
-pub fn create_pkg_dir(path: &str) -> result::Result<(), Error> {
-    let step = format!(
-        "{} {}Creating a pkg directory...",
-        style("[3/7]").bold().dim(),
-        emoji::FOLDER
-    );
-    let pb = PBAR.message(&step);
+pub fn create_pkg_dir(path: &str, step: &Step) -> result::Result<(), Error> {
+    let msg = format!("{}Creating a pkg directory...", emoji::FOLDER);
+    let pb = PBAR.step(step, &msg);
     let pkg_dir_path = format!("{}/pkg", path);
     fs::create_dir_all(pkg_dir_path)?;
     pb.finish();
     Ok(())
 }
 
-fn init(
-    path: Option<String>,
+struct Init {
+    crate_path: String,
     scope: Option<String>,
+    skip_build: bool,
     disable_dts: bool,
     target: String,
-    log: &Logger,
-) -> result::Result<(), Error> {
-    let started = Instant::now();
-
-    let crate_path = set_crate_path(path);
-
-    info!(&log, "Adding wasm-target...");
-    build::rustup_add_wasm_target()?;
-    info!(&log, "Adding wasm-target was successful.");
-
-    info!(&log, "Building wasm...");
-    build::cargo_build_wasm(&crate_path)?;
-
-    #[cfg(not(target_os = "windows"))]
-    info!(
-        &log,
-        "wasm built at {}/target/wasm32-unknown-unknown/release.", &crate_path
-    );
-    #[cfg(target_os = "windows")]
-    info!(
-        &log,
-        "wasm built at {}\\target\\wasm32-unknown-unknown\\release.", &crate_path
-    );
+    crate_name: Option<String>,
+}
 
-    info!(&log, "Creating a pkg directory...");
-    create_pkg_dir(&crate_path)?;
-    info!(&log, "Created a pkg directory at {}.", &crate_path);
+impl Init {
+    pub fn new(
+        path: Option<String>,
+        scope: Option<String>,
+        skip_build: bool,
+        disable_dts: bool,
+        target: String,
+    ) -> Init {
+        Init {
+            crate_path: set_crate_path(path),
+            scope,
+            skip_build,
+            disable_dts,
+            target,
+            crate_name: None,
+        }
+    }
+    pub fn process(&mut self, log: &Logger) -> result::Result<(), Error> {
+        let steps: Vec<fn(&mut Init, &Step, &Logger) -> result::Result<(), Error>> =
+            if self.skip_build {
```

Comment:
(ugh GH didn't post this comment in my original review)

so- i think we might be exploring even more "build" modes (see #153 ) - so (and i am likely overengineering here) but perhaps an enum and a match statement would work best here. would love to hear your thoughts!

Reply:
I did the modification this modification in the PR, if it is not like you what you had in mind I will revert it.